geode-core/src/test/java/org/apache/geode/cache30/CacheXmlNotFoundUnitTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache30;
+
+import org.apache.geode.cache.CacheFactory;
+import org.apache.geode.cache.CacheXmlException;
+import org.apache.geode.distributed.ConfigurationProperties;
+import org.apache.geode.test.junit.categories.UnitTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.io.File;
+import java.util.Properties;
+import java.util.Scanner;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.fail;
+
+@Category(UnitTest.class)
+public class CacheXmlNotFoundUnitTest {
+
+  /**
+   * unit test for <a href="https://issues.apache.org/jira/browse/GEODE-3038">GEODE-3038</a> Tests
+   * that an error about missing cache-xml file is indeed printed in the text log file. The test
+   * {@link org.apache.geode.cache30.CacheXml66DUnitTest#testNonExistentFile()} is supposed to test
+   * the same, but is not enough. It only checks for an CacheXmlException exception to be thrown.
+   * Also in that test a log is printed into STDOUT, and we do see our error there, but that is not
+   * the case when we work with the real text log, specified via "log-file" param.
+   */
+  @Test
+  public void testCacheXmlNotFoundInRealLog() throws Exception {
+
+    String CACHE_SERVER_LOG = "cacheXmlNotFoundUnitTest.log";
```

Comment:
In think this should be an IntegrationTest since it creates a real Cache and log file.

Reply:
Done, see 55a5cc982a56dc5ea160014b78ecc7ddf4961873